cmd/query/app/querysvc/query_service_v2.go

```diff
@@ -0,0 +1,174 @@
+func (qs QueryServiceV2) GetTraces(ctx context.Context, traceIDs ...tracestore.GetTraceParams) iter.Seq2[[]ptrace.Traces, error] {
+	getTracesIter := qs.traceReader.GetTraces(ctx, traceIDs...)
+	return func(yield func([]ptrace.Traces, error) bool) {
+		foundTraceIDs := make(map[pcommon.TraceID]struct{})
+		getTracesIter(func(traces []ptrace.Traces, err error) bool {
+			if err != nil {
+				return yield(nil, err)
+			}
+			for _, trace := range traces {
+				resources := trace.ResourceSpans()
+				for i := 0; i < resources.Len(); i++ {
+					scopes := resources.At(i).ScopeSpans()
+					for j := 0; j < scopes.Len(); j++ {
+						spans := scopes.At(j).Spans()
+						for k := 0; k < spans.Len(); k++ {
+							span := spans.At(k)
+							foundTraceIDs[span.TraceID()] = struct{}{}
+						}
+					}
+				}
+			}
+			return yield(traces, nil)
+		})
+		if qs.options.ArchiveTraceReader != nil {
+			missingTraceIDs := []tracestore.GetTraceParams{}
+			for _, id := range traceIDs {
+				if _, found := foundTraceIDs[id.TraceID]; !found {
+					missingTraceIDs = append(missingTraceIDs, id)
+				}
+			}
+			if len(missingTraceIDs) > 0 {
+				qs.options.ArchiveTraceReader.GetTraces(ctx, missingTraceIDs...)(yield)
+			}
+		}
+	}
+}
+
+// GetServices is the queryService implementation of tracestore.Reader.GetServices
+func (qs QueryServiceV2) GetServices(ctx context.Context) ([]string, error) {
+	return qs.traceReader.GetServices(ctx)
+}
+
+// GetOperations is the queryService implementation of tracestore.Reader.GetOperations
+func (qs QueryServiceV2) GetOperations(
+	ctx context.Context,
+	query tracestore.OperationQueryParameters,
+) ([]tracestore.Operation, error) {
+	return qs.traceReader.GetOperations(ctx, query)
+}
+
+// FindTraces is the queryService implementation of tracestore.Reader.FindTraces
+func (qs QueryServiceV2) FindTraces(ctx context.Context, query tracestore.TraceQueryParams) iter.Seq2[[]ptrace.Traces, error] {
+	return qs.traceReader.FindTraces(ctx, query)
+}
+
+// ArchiveTrace is the queryService utility to archive traces.
+func (qs QueryServiceV2) ArchiveTrace(ctx context.Context, traceID pcommon.TraceID) error {
+	if qs.options.ArchiveTraceWriter == nil {
+		return errNoArchiveSpanStorage
+	}
+	getTracesIter := qs.GetTraces(ctx, tracestore.GetTraceParams{TraceID: traceID})
+	var archiveErr error
+	getTracesIter(func(traces []ptrace.Traces, err error) bool {
+		if err != nil {
+			archiveErr = err
+			return false
+		}
+		for _, trace := range traces {
+			err = qs.options.ArchiveTraceWriter.WriteTraces(ctx, trace)
+			if err != nil {
+				return false
+			}
+		}
+		return true
+	})
+	return archiveErr
+}
+
+// Adjust applies adjusters to the trace.
+func (qs QueryServiceV2) Adjust(tracesIter iter.Seq[[]ptrace.Traces]) {
+	tracesIter(func(traces []ptrace.Traces) bool {
+		for _, trace := range traces {
+			qs.options.Adjuster.Adjust(trace)
+		}
+		return true
+	})
+}
```

Comment:
@yurishkuro Did I understand correctly in that this is what we wanted here? Or did you mean that we should change the underlying adjusters themselves to work on Seq?

Reply:
yes, you understood correctly. However, because adjusting needs to re-arrange the data I think this func should return `Seq[ptrace.Traces]` where each item is fully adjusted trace.